Assets/Scripts/Managers/AudioManager.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Services;
+using UnityEngine.UI;
+
+public class AudioManager : MonoService
+{
+    
+    // --- Sounds for the game ---
+    // UI
+    public AudioClip PauseButtonSound;
+    public AudioClip UiButtonPressed;
+    
+    // Gameplay 
+    public AudioClip BackgroundMusic;
+    public AudioClip FocusTileSound;
+    public AudioClip ReturnCameraTopDownSound;
+    public AudioClip SuccessfulSound;
+    public AudioClip FailSound;
+    
+    // --- AudioSource for playing audios
+    public GameObject AudioObject;
+    private AudioSource _soundEffectAudioSource;
+    private AudioSource _backgroundMusicAudioSource;
+    
+    // -- Service Initialization
+    private bool _isInitialized = false;
+    
+    // --- Buttons
+    // public Button PauseButton;
+    // public List<Button> UIButtons;
+    
+    // --- Member methods & functions ---
+    private void Start()
+    {
+        if (!AudioObject)
+            return;
+
+        // Find AudioSources for playing clips
+        _soundEffectAudioSource = AudioObject.transform.Find("SoundEffectAudio").GetComponent<AudioSource>();
+        _backgroundMusicAudioSource = AudioObject.transform.Find("BackgroundAudio").GetComponent<AudioSource>(); 
+        
+        // Initialize button listeners
+        // foreach (var button in UIButtons)
+        // {
+        //     button.onClick.AddListener(PlayUIButtonSFX);
+        //     Debug.Log($"AudioManager: Button from game object #{button.gameObject.name} added as UI button.");
+        // }
+
+        // Set Pause Button sound
+        // if (PauseButton)
+        // {
+        //     PauseButton.onClick.AddListener(PlayPauseButtonSFX);
+        // }
+        // else
+        // {
+        //     Debug.LogWarning("There is no pause button assigned to the AudioManager.");
+        // }
+        
+        // Camera management sounds
+        if (FocusTileSound)
+            ServiceLocator.GetService<CameraManager>().OnCameraFocus += PlayFocusTileSFX;
+        if (ReturnCameraTopDownSound)
+            ServiceLocator.GetService<CameraManager>().OnCameraTopDown += TopDownCameraSFX;
+        
+        // Init background music
+        _backgroundMusicAudioSource.clip = BackgroundMusic;
+        _backgroundMusicAudioSource.Play();
+        
+        _isInitialized = true;
+    }
+
+    public void PlaySuccessSFX()
+    {
+        PlaySFX(SuccessfulSound);
+    }
+
+    public void PlayFailSFX()
+    {
+        PlaySFX(FailSound);
+    }
+
+    public void PlayUIButtonSFX()
+    {
+        PlaySFX(UiButtonPressed);
+    }
+
+    public void PlayPauseButtonSFX()
+    {
+        PlaySFX(PauseButtonSound);
+    }
+
+    private void PlayFocusTileSFX(BoardSquare tile)
+    {
+        PlaySFX(FocusTileSound);
+    }
+
+    private void TopDownCameraSFX()
+    {
+        PlaySFX(ReturnCameraTopDownSound);
+    }
```

Comment:
Many of this should _really_ be in a single function, rather than a whole hardcoded lot of functions that all have a similar signature.

I am thinking of maintaining a private Unity-serialized array along with a private preinitialized `IReadOnlyDictionary<string, AudioClip>` or an enum, and export a method taking a string or an integer as a parameter.

Reply:
As for `PlayFocusTileSFX`, we could add an overload to that method adding an option on the stereo pan, or else.

Reply:
I fixed this issue. branch should be ready to merge